### ISSUE TITLE
About FxERC20Transfer withdraw test code 

### ISSUE THREAD

**Issue by picardcapt1212** [jdkanani/fx-portal#50]
Hi, i have a question about FxERC20Transfer withdraw testcode in `feature/hardhat_unittest`branch.
So i tested erc20 state transfer with `test/tunnel/FxERC20Tunnel.ts`.
I understand `mapToken`, `deposit` function, but i'm little confused of this [withdraw line](https://github.com/fx-portal/contracts/blob/b94102a343a8bb5dfb0997f0ad30843351d067c9/test/tunnel/FxERC20Tunnel.ts#L195).

In the test code after deposit(), the wallet.getAddress() has 10(i will omit 10^18) in child token contract(the mapped contract). 
And also the wallet.getAddress() has 9990 in the root token contract. 
```
child token contract wallet.getAddress() balance : 10
root token contract wallet.getAddress() balance : 9990
```

</br>

When withdraw() function call(withdraw 5 from child token contract) i thought below will be the result.
```
child token contract wallet.getAddress() balance : 5
root token contract wallet.getAddress() balance : 9995
```

But instead of above balance, the test code is showing me below result.
```
child token contract wallet.getAddress() balance : 5
root token contract wallet.getAddress() balance : 9990
```

So to sum up my question is
Q1. Is the wallet.getAddress() balance I expected right? - child token contract: 5, root token contract: 9995
Q2. Is the withdraw test code resulted becuase the checkpointManager is not working in local test environment?
Q3. What is the checkpointManager`0x600e7E2B520D51a7FE5e404E73Fb0D98bF2A913E` in the test code? [over here](https://github.com/fx-portal/contracts/blob/b94102a343a8bb5dfb0997f0ad30843351d067c9/test/shared/fixtures.ts#L104).

Thanks for advanced!

**Comment by wschwab** [comment]
We're working on a complete overhaul of the tests atm, we'll try to circle back to this once it's posted, and point to the code we think should answer the questions

**Comment by DhairyaSethi** [comment]
Withdrawal is a two step process, ie burn (`withdraw()` on child tunnel burns the token) and exit (submit exit-proof at `receiveMessage()` on root tunnel) which then unlock tokens on root. The tests have been updated to include the exit call (including proof generation) [here](https://github.com/0xPolygon/fx-portal/blob/main/hardhat/tunnel/FxERC20Tunnel.ts#L255). We've also migrated the test suite to dual of Foundry & Hardhat, more extensive cases covered in the Foundry unit tests [here](https://github.com/0xPolygon/fx-portal/blob/main/test/tunnel/FxERC20Tunnel.t.sol#L201). 
Do the checkout the updated project README for more info, & let me know if you have any questions!
